integration/v3_grpc_test.go

```diff
@@ -456,13 +456,12 @@ func TestV3TxnCmpHeaderRev(t *testing.T) {
 	for i := 0; i < 10; i++ {
 		// Concurrently put a key with a txn comparing on it.
 		revc := make(chan int64, 1)
+		errCh := make(chan error, 1)
 		go func() {
 			defer close(revc)
 			pr := &pb.PutRequest{Key: []byte("k"), Value: []byte("v")}
 			presp, err := kvc.Put(context.TODO(), pr)
-			if err != nil {
-				t.Fatal(err)
-			}
+			errCh <- err
```

Comment:
if `err != nil`, could `presp.Header.Revision` panic?

Reply:
Thanks for the catch! Yes, according to `(*kVClient).Put()`, `presp.Header.Revision` will panic. I changed my commit. Now it will return if `err != nil`

Reply:
If this go routine returns under err != nil, does the main go routine block at line 489?

Reply:
I don't think so. The child goroutine has a `defer close(revc)`. So if returns under `if err != nil`, the main goroutine will not block at `prev := <-revc`, and the main goroutine will call `t.Fatal(err)` at the next If instruction. In this case, `prev` will be 0, but this doesn't matter anyway